### PR TITLE
PE-387 Add custom_report_format method

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This branch adds the method custom_report format, that allows to return the answer for the given user and block.

### CODE REVIEW

File: freetextresponse/freetextresponse.py [hunk 1]
```diff
@@ -760,6 +759,35 @@ def save_reponse(self, data, suffix=''):
...
+        This method return the summited answer for the give user and block.
+        """
+        user = kwargs.get('user')
+        block = kwargs.get('block')
+
+        if block and user:
+            student_item_dict = block.student_item_key(user=user)
+            submission = sub_api.get_submissions(student_item_dict, limit=1)
+            try:
+                return submission[0]['answer']
```
Comment (Squirrel18) [inline freetextresponse/freetextresponse.py]: Could you please do a safe check here for 0 index and 'answer' key? @andrey-canon

File: freetextresponse/freetextresponse.py [hunk 2]
```diff
@@ -760,6 +759,35 @@ def save_reponse(self, data, suffix=''):
...
+            except IndexError:
+                pass
+
+            field = 'student_answer'
```
Comment (Squirrel18) [inline freetextresponse/freetextresponse.py]: Is this a constant? Please move it to a normal module constant. @andrey-canon

File: freetextresponse/freetextresponse.py [hunk 3]
```diff
@@ -760,6 +759,35 @@ def save_reponse(self, data, suffix=''):
...
+            descriptor = modulestore().get_item(block.location, depth=1)
+            field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+                block.location.course_key,
+                user,
+                descriptor,
+                asides=XBlockAsidesConfig.possible_asides(),
+            )
+            student_data = KvsFieldData(DjangoKeyValueStore(field_data_cache))
+            if student_data.has(block, field):
```
Comment (Squirrel18) [inline freetextresponse/freetextresponse.py]: Please, add a new break line between this and above line. @andrey-canon

File: freetextresponse/freetextresponse.py [hunk 4]
```diff
@@ -760,6 +759,35 @@ def save_reponse(self, data, suffix=''):
...
+                return student_data.get(block, field)
```
Comment (Squirrel18) [inline freetextresponse/freetextresponse.py]: Is this returning a string value? @andrey-canon

File: freetextresponse/freetextresponse.py [hunk 5]
```diff
@@ -760,6 +759,35 @@ def save_reponse(self, data, suffix=''):
         }
         return result
 
+    @staticmethod
+    def custom_report_format(*args, **kwargs):
+        """
+        This method return the summited answer for the give user and block.
```
Comment (Squirrel18) [inline freetextresponse/freetextresponse.py]: I think that the name of the function and its description doesn't match, could you please change one of them?

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: freetextresponse/freetextresponse.py (hunk 4 above)
Comment (andrey-canon) [reply freetextresponse/freetextresponse.py]: yes

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**